brian2genn/device.py

```diff
@@ -1203,9 +1203,11 @@ def process_poisson_groups(self, objects, poisson_groups):
             neuron_model.N = obj.N
             self.add_array_variables(neuron_model, obj)
             support_lines = []
-            suffix = '_thresholder';
-            lines = neuron_model.thresh_cond_lines;
-            codeobj = objects[obj.name + suffix].codeobj
+            for contained_obj in obj.contained_objects:
+                if isinstance(contained_obj, Thresholder):
+                    codeobj = contained_obj.codeobj
+                    break
```

Comment:
Since this is a `PoissonGroup`, there should be a single "spike" `Thresholder` object that we can access directly:
```suggestion
            codeobj = obj.thresholder['spike'].codeobj
```

Reply:
I sleepwalked over that one, nice catch.